app/vmagent/remotewrite/client.go

```diff
@@ -295,7 +296,7 @@ func (c *client) runWorker() {
 	ch := make(chan bool, 1)
 	for {
 		block, ok = c.fq.MustReadBlock(block[:0])
-		if !ok {
+		if !ok || len(block) == 0 {
```

Comment:
cmiiw this will exit the `worker` goroutine?

Reply:
Yes, `worker` goroutine will exit here. Did I misunderstand the meaning of "skipping"? It should be "continue" rather than "return"?

Reply:
```suggestion
		if !ok {
			return
		}
		if len(block) == 0 {
			logger.Warnf("remote-write worker skip block with size 0")  // The log here is optional since it does not help with solving the issue. It only indicates that the issue happens.
			continue
		}
```